### DISCUSSION TITLE
Facing some issue with useLazyAsyncData

### DISCUSSION THREAD

**Discussion by Croodo** [nuxt/nuxt#18422] [Questions] [answered]
Getting error when using **useLazyAsyncData** 
When navigating through NuxtLink 'title is set as post.value.title'  perfectly but on refresh can't set the title (on refresh title is set to **_undefined_**). What is wrong with this code? 

Actually, I'm trying to use the data outside watch so I can set head meta and do other other stuff

```
const { pending, data: post } = useLazyAsyncData("post", () => $fetch(uri));
const title = ref();
watch(post, () => {
  title.value = post.value.title;
});
useHead({
  title,
});
```

**Comment by Baroshem** [comment]
Hey,

If I understand the issue correctly, this is actually a correct behavior of `useLazyAsyncData` -> https://nuxt.com/docs/api/composables/use-lazy-async-data

```vue

<template>
  <div>
    {{ pending ? 'Loading' : count }}
  </div>
</template>
<script setup>
/* Navigation will occur before fetching is complete.
  Handle pending and error states directly within your component's template
*/
const { pending, data: count } = useLazyAsyncData('count', () => $fetch('/api/count'))
watch(count, (newCount) => {
  // Because count starts out null, you won't have access
  // to its contents immediately, but you can watch it.
})
</script>

```

**Comment by oritwoen** [comment]
As @Baroshem rightly stated `Lazy` fetched data only after loading the server. This means that initially your variable will be undefined.

In my opinion, it's better to use `useFetch` here, below is a sample code. The data will then be loaded immediately after refreshing the page and change after moving to another subpage.

```vue
<script setup lang="ts">
const route = useRoute()
const { data: post } = await useFetch(`post?slug=${route.path}`);

useHead({  
	title: post.value.title
})
</script>
```

**Comment by TheAlexLichter** [comment, marked as answer]
I think the best here would be to use `await useLazyAsyncData` (or `await useLazyFetch`). This will ensure that the title is available on the server-side and keep the desired client-side behavior.

**Reply by Croodo** [reply]
Holly Shit is just about the **_await_** , LOL 🤣. **Thanks Man** . 🙂
Spent a long time trying to find the solution.

**Reply by TheAlexLichter** [reply]
No problem 🙌🏻

**Reply by andrescastillodf** [reply]
I had a similar issue trying to se usehead inside useLazyAsyncData giving me nuxt instance unavailable randomly. 

I have solved the issue thanks to this comment and using watch with immediate: true 

I also have found that I need to return  an string instead of null on useLazyAsyncData to prevent this call to refresh on the client and showing the loading on the refresh.

```ts
const routePath = route.params.path as string
const { pending, data: page } = await useLazyAsyncData('pageData', async () => {
  return await getCmsPage(routePath)
    .then((res) => {
      return res;
    })
    .catch(() => {
      return ''
    });
})

watch(
  () => routePage.value,
  (newPage) => {
    if (newPage && typeof newPage === 'object') {
      useHead(generateSeoHead(newPage.Metadata || {} as Metadata, newPage.OpenGraph || {} as OpenGraph));
    } else if(!pending.value){
      useHead(() => ({
        status: 404,
        title: 'Error loading page',
      }));
    }
  },
  { immediate: true }
);
```

thanks @manniL and @Croodo
